Source/JavaScriptCore/runtime/JSGlobalObjectInlines.h

```diff
@@ -319,6 +319,56 @@ inline JSArray* constructArrayNegativeIndexed(JSGlobalObject* globalObject, Arra
     return ArrayAllocationProfile::updateLastAllocationFor(profile, array);
 }
 
+ALWAYS_INLINE JSArray* tryCreateContiguousArrayWithPattern(JSGlobalObject* globalObject, JSString* pattern, size_t initialLength)
+{
+    if (initialLength >= MIN_ARRAY_STORAGE_CONSTRUCTION_LENGTH)
+        return nullptr;
+
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    Structure* structure = globalObject->originalArrayStructureForIndexingType(ArrayWithContiguous);
+    RETURN_IF_EXCEPTION(scope, { });
```

Comment:
You do not need to have `RETURN_IF_EXCEPTION` for `originalArrayStructureForIndexingType`.